ros/api/v0/hosts.py

```diff
@@ -101,11 +102,20 @@ def get(self):
 
         account_query = db.session.query(RhAccount.id).filter(RhAccount.account == ident['account_number']).subquery()
 
-        if filter_display_name:
+        if filter_display_name and filter_by_state:
+            system_query = db.session.query(System.id)\
+                .filter(System.display_name.ilike(f'%{filter_display_name}%'))\
+                .filter(System.account_id.in_(account_query))\
+                .filter(System.state.ilike(f'%{filter_by_state}%'))
+        elif filter_display_name:
             system_query = db.session.query(System.id)\
                 .filter(System.display_name.ilike(f'%{filter_display_name}%'))\
                 .filter(System.account_id.in_(account_query))\
                 .filter(System.state.in_(SYSTEM_STATES_EXCEPT_EMPTY))
+        elif filter_by_state:
+            system_query = db.session.query(System.id)\
+                .filter(System.state.ilike(f'%{filter_by_state}%'))\
+                .filter(System.account_id.in_(account_query))
         else:
             system_query = db.session.query(System.id)\
                 .filter(System.account_id.in_(account_query))\
```

Comment:
Thank you @r14chandra for adding state filter.
Functionality wise looks good but I think we need to refactor this `if..elif..else` logic.
Currently, we just have three filters - account, display_name & state, think about adding more filters. 
@r14chandra -  To make it easier to add new filter as well as more readable, it would be great if you could refactor this code block.